app/services/foreman_deprecation.rb

```diff
@@ -0,0 +1,24 @@
+class ForemanDeprecation
+  def self.deprecation_warning(moved_to_controller, moved_to_method, foreman_version_deadline, additional_info = nil,callstack = caller)
+    raise Foreman::Exception, "Invalid version format, please enter in x.y.z" unless foreman_version_deadline.to_s.match(/^\d[.]\d\d?[.]\d$/)
+    msg = "You are using deprecated behavior, it will be removed in version #{foreman_version_deadline}, use #{moved_to_controller}##{moved_to_method} instead.\n"
+    msg += additional_info unless additional_info.nil?
+    ActiveSupport::Deprecation.warn(msg,callstack)
```

Comment:
This normally logs the last piece of the stack trace, which is going to be of limited use here, since it's always going to be ForemanDeprecation.  Should we do our own logging?